content/root/best-practices-guide.textile

```diff
@@ -0,0 +1,97 @@
+* Want to set up realtime communication with IoT devices on platforms for which there's no native Ably client library available? Use the "MQTT protocol adapter":/mqtt.
+* Want to migrate from PubNub/Pusher? Use the relevant "protocol adapter":/adapters to "ease the pain of migration":https://support.ably.io/solution/articles/3000054127-can-you-help-ease-the-pain-of-migration-from-another-realtime-service: you can switch to Ably right away while sticking with your current client libraries, and gradually migrate to native Ably clients.
+
+h4. 8. Subscribing to events on the server&#45;side
+
+* Subscribing to events on the server side using the standard Pub/Sub pattern can be a bad idea for a number of reasons as explained in this "blog article":https://blog.ably.io/message-queues-the-right-way-to-process-and-work-with-realtime-data-on-your-servers-2d15985301f8. With Message Queues, you can set up multiple worker servers to distribute the load of the messages received from Ably while ensuring each of the messages gets processed exactly once by any one of your worker servers.
+
+h4. 9. WebHooks/Reactor Functions vs. Message Queues &#45; when to use what?
+
+* "WebHooks":/general/webhooks and "Reactor Functions":/general/functions work well for low or medium message volumes, but "Reactor Queues":/general/queues will scale better to high message volumes. For very high volumes, we recommend using "Reactor Firehose":/general/firehose.
+
+h4. 10. Multiplexing channels
+
+* You should almost always instantiate a single Ably client library instance per web page or server. Don't forget that the Client Library SDK can "multiplex":https://support.ably.io/support/solutions/articles/3000058406-do-you-support-channel-groups-and-multiplexing many channels over a single connection.
+
+h4. 11. Limiting the effect of a security compromise using the principle of least privilege
+
+* Using the available "authentication methods":/general/authentication in Ably, you can choose the permissions or capabilities to be given to a client. You can do this by restricting the permissions on the API key as well as restricting the permissions your auth server grants to that API key whilst granting a token request. You can restrict the channels that a client can access (including with wildcards), as well as the operations the client can perform within those channels, using "capabilities":/general/authentication#capabilities-explained.
+* Always make sure to limit the capabilities given to a client to only what's required by the client to do their job. This prevents any unexpected behaviour by a client and increases the security of your application. This is called the principle of least privilege.
+
+h4. 12. Efficient use of channels
+
+Channels consume resources and hence cost you money. Make sure that you are using them efficiently and in accordance to the limit you have as per your package. However, bear in mind the following:
```

Comment:
I think we should try and explain that whilst we recommend they use channels to distribute work more evenly across the cluster, there is an associated cost for a high number of channels, so common sense must prevail in regards to channel use. Here are some considerations that may help you with your channel usage design: